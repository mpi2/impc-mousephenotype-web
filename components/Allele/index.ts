export { default as AlleleMap } from './AlleleMap';
export { default as Crispr } from './Crispr';
export { default as ESCell } from './ESCell';
export { default as IVP } from './IVP';
export { default as Mice } from './Mice';
export { default as QCModal } from './QCModal';
export { default as TVP } from './TVP';
