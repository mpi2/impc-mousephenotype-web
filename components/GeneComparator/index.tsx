import { useRouter } from "next/router";
import React, { useEffect, useState } from "react";
import { Button } from "react-bootstrap";

const GeneComparatorContext = React.createContext(null);

export const GeneComparatorProvider = ({ children }) => {
  const [genes, setGenes] = useState([]);

  const addGene = (gene) => {
    if (gene && !genes.includes(gene)) {
      const newGenes = [...genes, gene];
      setGenes(newGenes);
      return newGenes;
    }
    return genes;
  };

  const removeGene = (gene) => {
    if (gene && genes.includes(gene)) {
      genes.splice(genes.indexOf(gene));
      setGenes(genes);
    }
  };

  const resetGenes = (genes: string[]) => {
    setGenes(genes);
  };

  return (
    <GeneComparatorContext.Provider
      value={{ genes, addGene, removeGene, resetGenes }}
    >
      {children}
    </GeneComparatorContext.Provider>
  );
};

export const useGeneComparator = () => React.useContext(GeneComparatorContext);

export const GeneComparatorTrigger = ({ current }: { current?: string }) => {
  const { addGene, genes } = useGeneComparator();
  const router = useRouter();
  return (
    <Button
      style={{
        position: "fixed",
        zIndex: 100,
        bottom: genes.length > 0 || current ? 0 : -100,
        right: 100,
        transition: ".3s ease transform",
      }}
      variant="primary"
      onClick={() => {
        if (current) {
          const updatedGenes = addGene(current);
          return router.push(`/data/compareGenes/?genes=${updatedGenes.join("_")}`);
        }
        router.push(`/data/compareGenes/?genes=${genes.join("_")}`);
      }}
    >
      Compare Genes{" "}
      <span
        className="bg-white"
        style={{
          display: "inline-block",
          height: 22,
          padding: "0 7px",
          borderRadius: 11,
          marginLeft: 6,
          fontWeight: "bold",
          color: '#000',
        }}
      >
        {genes.length}
      </span>
    </Button>
  );
};
