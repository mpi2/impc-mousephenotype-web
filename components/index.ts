export { default as Card } from './Card';
export { default as Search } from './Search';
export { default as AlleleSymbol } from './AlleleSymbol';
export { default as DownloadData } from './DownloadData'