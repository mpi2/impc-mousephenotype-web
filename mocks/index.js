export async function initMocks() {
  if (typeof window === "undefined") {
    const { server } = await import("./server");
    server.listen();
  } else {
    const { worker } = await import("./browser");
    return worker.start({
      serviceWorker: {
        url: "/data/mockServiceWorker.js",
        options: {
          scope: "/",
        }
      }
    });
  }
}

