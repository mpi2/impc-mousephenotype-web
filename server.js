const { createServer } = require('http');
const { parse } = require('url');
const next = require('next');

const dev = process.env.NODE_ENV !== 'production';
const hostname = 'localhost';
const port = 3000;
const app = next({ dev, hostname, port });
const handle = app.getRequestHandler();

app.prepare()
.then(() => {
    createServer(async (req, res) => {
        try {
            const parsedUrl = parse(req.url, true);
            const { pathname, query} = parsedUrl;
            const host = req.headers.host;
            console.log({ host, pathname });
            res.setHeader('X-IMPC-TEST', "true");
            if (host.includes('nginx.mousephenotype-dev.org')) {
                const modifiedPathName = pathname.replace("/data/", "");
                console.log({ modifiedPathName });
                await app.render(req, res, modifiedPathName, query);
                res.setHeader('X-PATH', JSON.stringify({ host, pathname, modifiedPathName }));
            } else {
                await handle(req, res, parsedUrl);
            }

        } catch (err) {
            console.error('Error occurred handling', req.url, err)
            res.statusCode = 500
            res.end('internal server error')
        }
    })
        .once('error', (err) => {
            console.error(err);
            process.exit(1);
        })
        .listen(port, () => {
            console.log(`Ready on http://${hostname}:${port}`);
        })
});