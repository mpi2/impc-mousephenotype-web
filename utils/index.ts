import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};
