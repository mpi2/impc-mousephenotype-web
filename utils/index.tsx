import React from "react";
import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};

export const formatPValue = (pValue: number) => {
  const pValueArray = Number.parseFloat(String(pValue))
    .toExponential(2)
    .split("e");
  return (
    <>
      {pValueArray[0]}x10<sup>{pValueArray[1].replace("+", "")}</sup>
    </>
  );
};

export const formatESCellName = (src: string) => {
  const [name, superscript, end] = src.split(/<|>/);
  return (
    <>
      {name}
      <sup>{superscript}</sup>
      {end}
    </>
  );
};

const fetchCache = {};
export { fetchCache };

export const toSentenceCase = (camelCase) => {
  if (camelCase) {
    const result = camelCase.replace(/([A-Z])/g, " $1");
    return result[0].toUpperCase() + result.substring(1).toLowerCase();
  }
  return "";
};
