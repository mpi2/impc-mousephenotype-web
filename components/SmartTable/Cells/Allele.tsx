import { Model, TableCellProps } from "@/models";
import _ from "lodash";
import { formatAlleleSymbol } from "@/utils";

const AlleleCell = <T extends Model>(props: TableCellProps<T>) => {
  const allele = formatAlleleSymbol(_.get(props.value, props.field) as string);
  return (
    <>
      {allele[0]}
      <sup>{allele[1]}</sup>
    </>
  )
};

export default AlleleCell;