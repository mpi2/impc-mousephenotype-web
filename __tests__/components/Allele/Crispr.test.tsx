import { render, screen, waitFor } from '@testing-library/react';
import { Crispr } from "@/components/Allele";
import { QueryClientProvider } from "@tanstack/react-query";
import { API_URL, createTestQueryClient } from "../../utils";
import mockRouter from "next-router-mock";
import { GeneComparatorProvider } from "@/components/GeneComparator";
import { server } from "../../../mocks/server";
import { rest } from "msw";
import dbn1Crispr from "../../../mocks/data/tests/dbn1-crispr.json";
import userEvent from '@testing-library/user-event';

jest.mock('next/router', () => jest.requireActual('next-router-mock'));

describe('Allele Page - Crispr', () => {
  it('renders correctly', async () => {
    const user = userEvent.setup();
    const endpointURL = `${API_URL}/api/v1/alleles/crispr/get_by_mgi_and_allele_superscript/MGI:1931838/em1(IMPC)Bay`;
    server.use(
      rest.get(endpointURL, (req, res, ctx) => {
        return res(ctx.status(200), ctx.json(dbn1Crispr));
      })
    );

    const client = createTestQueryClient();
    await mockRouter.push('/data/alleles/MGI:1931838/em1(IMPC)Bay');

    document.execCommand = jest.fn();

    const { container } = render(
      <QueryClientProvider client={client}>
        <GeneComparatorProvider>
          <Crispr
            mgiGeneAccessionId="MGI:1931838"
            alleleName="em1(IMPC)Bay"
          />
        </GeneComparatorProvider>
      </QueryClientProvider>
    );
    await waitFor(() => expect(screen.getByTestId('header')).toHaveTextContent('Sequence'))
    expect(container).toMatchSnapshot();

    await user.click(screen.getByRole('button', { name: "Copy"}));
    expect(document.execCommand).toHaveBeenCalledWith("copy");
  });
});