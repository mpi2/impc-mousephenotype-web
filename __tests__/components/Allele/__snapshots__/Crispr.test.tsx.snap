// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Allele Page - Crispr renders correctly 1`] = `
<div>
  <div
    class="card"
    id="crispr"
  >
    <h2
      data-testid="header"
    >
      Sequence
    </h2>
    <div
      class="bg-grey-light false"
      style="padding: 1rem; position: relative;"
    >
      &gt;Dbn1 Cas9_Dbn1_13790
GCTAGGGAAGGGGCCTAGTAGGGTGGTCCAGCTCAGTGCTATTCTTGGAGCTGTGAGATTTCCTTAAGCAAAGCTTCCTGTACCTTGCTGGGCACTAGCACCTGGAAAGACACGAGAGGTGTGATCCCCGAGTTGTGGGTTCCTGGTGTTTGTGTGAAAAACAAACAGACAAATAACAGCAGCAGCAACAACAAAACATTTTTTTTCGACGTTTTTCTCCAAATTGGAAAAGAAAATCCACCTTGACTCAGACCCTCGCTGACCTGGCTCCTCCACTGCTTCCGCCTTGAGACCTGGGGTCCTTGGGGTGTATTAGTCTTGGGCTCTGCATTTG

      <p
        class="grey mt-2 mb-0 small"
      >
        Type: 
        complete
        <span
          class="ms-3 me-3 "
        >
          |
        </span>
        Category:
         
        outcome sequence
      </p>
      <div
        style="position: absolute; bottom: 16px; right: 16px;"
      >
        <button
          class="btn impc-secondary-button small"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-copy "
            data-icon="copy"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M208 0H332.1c12.7 0 24.9 5.1 33.9 14.1l67.9 67.9c9 9 14.1 21.2 14.1 33.9V336c0 26.5-21.5 48-48 48H208c-26.5 0-48-21.5-48-48V48c0-26.5 21.5-48 48-48zM48 128h80v64H64V448H256V416h64v48c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48z"
              fill="currentColor"
            />
          </svg>
           
          Copy
        </button>
      </div>
    </div>
  </div>
  <div
    class="card"
  >
    <h2>
      Crispr details
    </h2>
    <p>
      <a
        class="link primary"
        href="https://www.informatics.jax.org/allele/MGI:6257430"
        target="_blank"
      >
        MGI:1931838
        Â 
        <span
          class="grey"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up-right-from-square fa-xs "
            data-icon="arrow-up-right-from-square"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32h82.7L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3V192c0 17.7 14.3 32 32 32s32-14.3 32-32V32c0-17.7-14.3-32-32-32H320zM80 32C35.8 32 0 67.8 0 112V432c0 44.2 35.8 80 80 80H400c44.2 0 80-35.8 80-80V320c0-17.7-14.3-32-32-32s-32 14.3-32 32V432c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V112c0-8.8 7.2-16 16-16H192c17.7 0 32-14.3 32-32s-14.3-32-32-32H80z"
              fill="currentColor"
            />
          </svg>
        </span>
      </a>
    </p>
    <h3
      class="mb-0 mt-1"
    >
      Nucleases
    </h3>
    <div
      class="mt-3"
    >
      <div>
        <span
          class="grey"
        >
          Type
          : 
        </span>
        <strong>
          Cas9
        </strong>
      </div>
      <div>
        <span
          class="grey"
        >
          Class
          : 
        </span>
        <strong>
          mRNA
        </strong>
      </div>
    </div>
    <h3
      class="mb-0 mt-4"
    >
      Genotype primers
    </h3>
    <div
      class="mt-3"
    >
      <div>
        <span
          class="grey"
        >
          Name
          : 
        </span>
        <strong>
          Geno_Phldb2_F1
        </strong>
      </div>
      <div>
        <span
          class="grey"
        >
          Sequence
          : 
        </span>
        <strong>
          CACTGGGCGCAAAATTCTTCT
        </strong>
      </div>
    </div>
    <h3
      class="mb-0 mt-4"
    >
      Guides
    </h3>
    <div
      class="tableWrapper"
    >
      <table
        class="table striped undefined table table-bordered"
      >
        <thead>
          <tr>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: bold;"
              >
                Guide sequence
                 
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-caret-up sortIcon"
                  data-icon="caret-up"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                PAM
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                CHR
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                Start
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                Stop
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                Strand
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                Genome build
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                GRNA concentration
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                Truncated guide
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                Reversed
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                Sanger service
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                Guide format
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              rowspan="2"
              width="16.666666666666664%"
            >
              <button
                class="inlineButton headerButton"
                style="font-weight: normal;"
              >
                Guide source
                 
                <span
                  class="defaultIcons"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up first"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down second"
                    data-icon="caret-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </th>
          </tr>
          <tr />
        </thead>
        <tbody>
          <tr>
            <td>
              GAGGTGTGATCCCCGAGAAA
            </td>
            <td>
              TGG
            </td>
            <td>
              13
            </td>
            <td>
              55630835
            </td>
            <td>
              55630857
            </td>
            <td>
              -
            </td>
            <td>
              GRCm39
            </td>
            <td>
              5
            </td>
            <td />
            <td />
            <td />
            <td />
            <td />
          </tr>
          <tr>
            <td>
              CAGGTCCAACACCTATCTTG
            </td>
            <td>
              TGG
            </td>
            <td>
              13
            </td>
            <td>
              55629036
            </td>
            <td>
              55629058
            </td>
            <td>
              -
            </td>
            <td>
              GRCm39
            </td>
            <td>
              5
            </td>
            <td />
            <td />
            <td />
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
